### PR TITLE
Clean up runtime errors, fix VueUIs opening offscreen

### LINKED ISSUE #11274: Body scanner console and sleeper UIs are broken
To be more specific, it doesn't fit on screens with low resolutions. I use somewhat like 1360x768 and the upper side is off-screen, so i can't move the ui nor close it. The only way to close it is to go far away enough. 
image
The round ID is cal-cC6Y. I talked to the admin but they didn't give me a proper answer (and i don't remember the ckey).
I think that the way to fix it is to make their UI a bit smaller.

### PR DESCRIPTION
**Pull request by JohnWildkins**
basically fixes a bunch of minor issues from the last two days of runtime errors, plus makes VueUIs clamp to screen size and ensures that away sites with the same name don't overwrite each other

changes:
  - bugfix: "Cleaned up several misc. runtime errors."
  - bugfix: "VueUI windows will now clamp to slightly less than the current BYOND window's screen height, hopefully preventing windows from being stuck open."
  - refactor: "Away sites with the same player-facing name no longer overwrite each other in site selection."

Fixes #11274

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!review

**Comment by JohnWildkins** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

File: code/modules/heavy_vehicle/interface/screen_objects.dm [hunk 1]
```diff
@@ -7,7 +7,7 @@
 	maptext_y = 11
 
 /obj/screen/mecha/proc/notify_user(var/mob/user, var/text)
-	if(user.loc == owner)
+	if(user?.loc == owner)
```
Comment (mikomyazaki) [inline code/modules/heavy_vehicle/interface/screen_objects.dm:10]:
```suggestion
	if(user && user.loc == owner)
```

File: code/modules/projectiles/gun.dm [hunk 2]
```diff
@@ -142,6 +142,8 @@
 
 	if(pin && needspin)
 		pin = new pin(src)
+	else if(pin)
+		pin = null
```
Comment (mikomyazaki) [inline code/modules/projectiles/gun.dm:146]: Can this just be new'd on line 141? It's kind of weird for the `pin` var to contain a path and a ref to a pin obj instance.

We should always have a pin on line 143 if `needspin == TRUE`, due to the check on line `140`.

File: code/modules/vueui/ui.dm [hunk 3]
```diff
@@ -90,7 +90,13 @@ main ui datum.
 
 	var/params = "window=[windowid];file=[windowid];titlebar=0;can_resize=0;"
 	if(width && height)
-		params += "size=[width]x[height];"
+		var/winsize = winget(user, "mapwindow", "size")
+		var/map_size = splittext(winsize, "x")
+		var/screen_height = text2num(map_size[1])
+		if(height > screen_height)
+			params += "size=[width]x[screen_height - 100];"
```
Comment (mikomyazaki) [inline code/modules/vueui/ui.dm:97]: What happens if we're for some reason playing with a screen that has <100 height? Does a negative height do something weird?

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

**Review by mikomyazaki — APPROVED** [review]

**Pull request merged**